### ISSUE TITLE
Bug: The requested module '/npm/@pixi/color@7.2.0-beta/+esm' does not provide an export named 'Color'

### ISSUE THREAD

**Issue by 8Observer8** [pixijs/pixijs#9189]
### Current Behavior

Error message: The requested module '/npm/@pixi/color@7.2.0-beta/+esm' does not provide an export named 'Color' 

### Expected Behavior

There should be no errors.

### Steps to Reproduce

- Go to https://www.jsdelivr.com/
- Type: pixi
- Select 7.2.0-beta
- Click on ESM button
- Copy this link: https://cdn.jsdelivr.net/npm/pixi.js@7.2.0-beta/+esm
- Use this link to set up Pixi.js with importmap:

index.html

```html
<!DOCTYPE html>

<html lang="en">

<head>
    <meta charset="UTF-8">
    <title>Example</title>
</head>

<body style="caret-color: transparent;">
    <canvas id="renderCanvas" width="400" height="400"></canvas>

    <!-- Since import maps are not yet supported by all browsers, it is
        necessary to add the polyfill es-module-shims.js
        Source: https://threejs.org/docs/index.html#manual/en/introduction/Installation -->
    <script async src="https://unpkg.com/es-module-shims@1.3.6/dist/es-module-shims.js"></script>
    <script type="importmap">
        {
            "imports": {
                "pixi-js": "https://cdn.jsdelivr.net/npm/pixi.js@7.2.0-beta/+esm",
                "matter-js": "https://cdn.jsdelivr.net/npm/matter-js@0.19.0/+esm"
            }
        }
    </script>

    <script type="module" src="js/main.js"></script>
</body>

</html>
```

js/main.js

```js
// import * as PIXI from "pixi-js";
import { default as PIXI } from "pixi-js";
import { default as Matter } from "matter-js";

console.log(`Matter.js version: ${Matter.version}`);

console.log(PIXI);
```

### Environment

- **`pixi.js` version**: *7.2.0-beta*
- **Browser & Version**: *Chrome 110*
- **OS & Version**: *Windows 10*
- **Running Example**: 
- https://playcode.io/1211309
- https://plnkr.co/edit/cpTRfBpoIdl2ZPto

**Comment by SuperSodaSea** [comment]
> https://cdn.jsdelivr.net/npm/@pixi/color@7.2.0-beta/+esm

```
/**
 * Failed to bundle using Rollup v2.74.1.
 * If you believe this to be an issue with jsDelivr, and not with the package itself, please open an issue at https://github.com/jsdelivr/jsdelivr
 */

 throw new Error('Failed to bundle using Rollup v2.74.1. If you believe this to be an issue with jsDelivr, and not with the package itself, please open an issue at https://github.com/jsdelivr/jsdelivr');
```

It looks strange...

**Comment by SuperSodaSea** [comment]
@bigtimebuddy Although it may not be related to this issue, it seems that we accidentally included files from `colord/plugins/names` in `7.2.0-beta` release (see <https://unpkg.com/browse/@pixi/color@7.2.0-beta/lib/external/colord/plugins/>). Maybe we need to update the `external` in rollup.config.js?

<https://github.com/pixijs/pixijs/blob/28bc3f775fbb28babb354dd57269dce09ac23f26/rollup.config.js#L87-L91>

**Comment by bigtimebuddy** [comment]
Update: I manually fixed the dist-tags so 7.2.0-beta.2 should be on `prerelease` and 7.1.2 on `latest`

**Comment by 8Observer8** [comment]
I created a new example on Plunker. I replaced this link `https://cdn.jsdelivr.net/npm/pixi.js@7.2.0-beta/+esm` by this `https://cdn.jsdelivr.net/npm/pixi.js@7.1.2/+esm` but I have the same error:

```
Uncaught SyntaxError: The requested module '/npm/@pixi/color@7.2.0-beta/+esm' does not provide an export named 'Color' (at VM427 +esm:7:250)
```

**Comment by SuperSodaSea** [comment]
> I created a new example on Plunker. I replaced this link `https://cdn.jsdelivr.net/npm/pixi.js@7.2.0-beta/+esm` by this `https://cdn.jsdelivr.net/npm/pixi.js@7.1.2/+esm` but I have the same error:
> 
> `Uncaught SyntaxError: The requested module '/npm/@pixi/color@7.2.0-beta/+esm' does not provide an export named 'Color' (at VM427 +esm:7:250)`

This is caused by #8798, `@pixi/` packages do not declare `peerDependencies` now, jsDelivr don't know which version of package to choose, so it use the `latest` version (which is cached as `7.2.0-beta`).

Now we can only wait for jsDelivr flusing its cache to make `@7.1.2` work (don't know how long does it take :cry:). For `@7.2.0-xxx` we need to wait for final release of `@7.2.0` to make it work.

**Comment by SuperSodaSea** [comment]
It seems that the cache has been flushed but `7.1.2` is still not working, because we [accidentally bundled `css-color-names` into `@pixi/utils`](https://unpkg.com/browse/@pixi/utils@7.1.2/lib/external/css-color-names/), with [invalid source map](https://unpkg.com/browse/@pixi/utils@7.1.2/lib/color/hex.mjs.map), causing <https://cdn.jsdelivr.net/npm/@pixi/utils@7.1.2/+esm> failed to bundle. So we can only wait for `7.2.0` now... :cry:

**Comment by SuperSodaSea** [comment]
It turns out that this (generating invalid source map) is a bug of [`rollup-plugin-rename-node-modules`](https://www.npmjs.com/package/rollup-plugin-rename-node-modules), see Lazyuki/rollup-plugin-rename-node-modules#10 for detail. I will try to fix that.

BTW I found that the `ismobilejs` is still bundled into `@pixi/settings@7.2.0-beta.2` and generating invalid source map, since it is marked as `devDependencies` instead of `dependencies`. I will also fix this.

**Comment by 8Observer8** [comment]
This version doesn't work: `https://cdn.jsdelivr.net/npm/pixi.js@7.1.2/+esm`

`
Uncaught SyntaxError: The requested module '/npm/@pixi/utils@7.1.3/+esm' does not provide an export named 'deprecation'
`

**Comment by 8Observer8** [comment]
I replaced `7.2.0-beta` to `7.1.2` (not `7.1.3`) and I have this error: https://github.com/pixijs/pixijs/issues/9210

`
Bug: The requested module '/npm/@pixi/utils@7.1.3/+esm' does not provide an export named 'deprecation'
`

**Running Example**: 
- https://playcode.io/1230828
- https://plnkr.co/edit/HjnXqDsq9uOJxjkW

**Comment by bigtimebuddy** [comment]
We resolved most of the blocking issues here with sourcemaps and ESM builds, but we aren't going to be able to fully support this until we fix #8798. Because we aren't declaring peers, it's trying to select the latest version which we don't want. Which in this case is wrong:

https://jsfiddle.net/bigtimebuddy/5ehL3f6q/
Trying to fetch `/npm/@pixi/utils@7.1.3/+esm` when it _should_ be `/npm/@pixi/utils@7.2.0-beta.3/+esm`

I'm currently leaning toward reverting the changes to peer dependencies and going back to declaring them (at the expense of having more mysterious version conflicts). Long-term, we have a better strategy for shipping pixi that circumvents all the peer issues, but that'll likely be v8.

**Comment by bigtimebuddy** [comment]
To elaborate a little more on my last point, we've been beta testing an alternative delivery of this project called [pixijs](https://www.npmjs.com/package/pixijs). This ships all of the packages in _one_ package. It circumvents all of the peer issues we've been having and is all round easier and more scaleable.

Also it works with jsdelivr + esm. Example:
https://jsfiddle.net/bigtimebuddy/dvw49je5/

This is confirmation that we should go this route for v8. Feel free to use `pixijs` package instead.

**Comment by MartinKolarik** [comment]
I wonder, had the declared peer dependencies caused any issues? We aim to make the bundling work in these cases, but if there's no version specified in `package.json` we obviously can't pick the correct one.

**Comment by SuperSodaSea** [comment]
@MartinKolarik It's a long story, you can see #8630 and #8798 for more information. In my opinion we should definitely bring peerDependencies back.

**Comment by bigtimebuddy** [comment]
This has been resolved with 7.2.0-rc:
https://jsfiddle.net/bigtimebuddy/5ehL3f6q/

**Comment by 8Observer8** [comment]
Thanks a lot guys! Unfortunately JSFiddle doesn't work in Russia (VPN only). But CodePen also supports `importmap`: https://codepen.io/8Observer8/pen/rNZmawW
